e3/sys.py

```diff
@@ -235,18 +235,44 @@ def python_script(name, prefix=None):
     :return: a list that will be the prefix of your command line
     :rtype: list[str]
     """
+    def has_relative_python_shebang(file_script):  # unix: no cover
+        """Return True if the script contains #!python shebang.
+
+        When producing relocatable python distribution we change the shebang
+        to #!python. In that case prefix the command line with the current
+        python interpreter.
+
+        #!/path/to/python shebang should return false, as we don't need to
+        return interpreter path.
+        """
+        with open(file_script, 'rb') as f:
+            content = f.read()
+        return re.search(b'#!python', content, flags=re.MULTILINE) is not None
+
     if prefix is None:
         if sys.platform == 'win32':  # unix: no cover
             prefix = os.path.dirname(sys.executable)
         else:
             prefix = os.path.dirname(os.path.dirname(sys.executable))
 
     if sys.platform == 'win32':  # unix: no cover
-        script = os.path.join(prefix, 'Scripts', name)
+        if os.path.basename(prefix) == 'scripts':  # defensive code
```

Comment:
I don't like seeing #defensive code on two branches. Can you rewrite that as:

script = os.path.join(prefix, name) if os.path.basename(prefix) == 'scripts' \
    else os.path.join(prefix, 'Scripts', name)